ci/bazel-scripts/targets.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+#
+#   targets.py [-h] [--skip_long_tests] [--commit_range COMMIT_RANGE] {build,test}
+#
+# This script is invoked from the CI Main workflow to print to stdout which Bazel targets should be built or tested.
+#
+# If --commit_range is passed only bazel targets will be included that have modified inputs within the specified git COMMIT_RANGE.
+#
+# If --skip_long_tests is passed, tests tagged with 'long_test' will be excluded.
+#
+# The script will print the bazel query to stderr which is useful for debugging:
+#   ci/bazel-scripts/targets.py --skip_long_tests --commit_range=master..HEAD test
+#   bazel query --keep_going '((kind(".*_test", //...)) except attr(tags, long_test, //...)) except attr(tags, manual, //...)'
+
+import argparse
+import fnmatch
+import subprocess
+import sys
+
+# Return all bazel targets (//...) sans the long_tests (if --skip_long_tests is specified)
+# in case any file is modified matching any of the following globs:
+all_targets_globs = ["*.bazel", "*.bzl", ".bazelrc", ".bazelversion", "mainnet-*-revisions.json", ".github/*"]
+
+
+def diff_only_query(command: str, commit_range: str, skip_long_tests: bool) -> str:
+    """
+    Return a bazel query for all targets that have modified inputs in the specified git commit range. Taking into account:
+    * To return all targets in case files matching all_targets_globs are modified.
+    * To only include test targets in case the bazel command was 'test'.
+    * To exclude long_tests if requested.
+    """
+    modified_files = subprocess.run(
+        ["git", "diff", "--name-only", commit_range], check=True, capture_output=True, text=True
+    ).stdout.splitlines()
+
+    # The files matching the all_targets_globs are typically not depended upon by any bazel target
+    # but will determine which bazel targets there are in the first place so in case they're modified
+    # simply return all bazel targets. Otherwise return all targets that depend on the modified files.
+    if any(len(fnmatch.filter(modified_files, glob)) > 0 for glob in all_targets_globs):
+        query = "//..."
+    else:
+        # Note that modified_files may contain files not depended upon by any bazel target.
+        # `bazel query --keep_going` will ignore those but will return the special exit code 3
+        # in case this happens which we check for below.
+        query = "rdeps(//..., set({targets}))".format(targets=" ".join(modified_files))
+
+    # The targets returned by this script will be passed to `bazel test` by the caller (in case there are any).
+    # We have to ensure that the targets include at least one test otherwise `bazel test` will error with:
```

Comment:
I think this is not 100% correct, right? at least they should not include _only_ build targets, but it may be an empty list